src/main/java/controller/login/LoginState.java

```diff
@@ -1,7 +1,35 @@
 package controller.login;
 
+/**
+ * The state for the Login View Model.
+ */
 public class LoginState {
-    public LoginState() {
-        
+    private String email = "";
+    private String loginError;
+    private String password = "";
```

Comment:
just some rlly minor changes, change loginError to emailError and passwordError, and add getters + setters